doc/src/manual/metaprogramming.md

```diff
@@ -11,6 +11,19 @@ code in Julia are represented by Julia data structures, powerful [reflection](ht
 capabilities are available to explore the internals of a program and its types just like any other
 data.
 
+!!! warning
+    Metaprogramming is a powerful tool, but it introduces complexity that can make code more
+    difficult to understand. For example, it can be surprisingly hard to get scope rules
+    correct. Metaprogramming should typically be used only when other approaches such as
+    [higher order functions](@ref man-anonymous-functions) and
+    [closures](https://en.wikipedia.org/wiki/Closure_(computer_programming)) cannot be applied.
+
+    `eval` and defining new macros should be typically used as a last resort. It is almost
+    never a good idea to use `Meta.parse` or convert an arbitrary string into julia code.
```

Comment:
```suggestion
    never a good idea to use `Meta.parse` or convert an arbitrary string into Julia code. For
    manipulating Julia code, use the `Expr` data structure directly to avoid having to     deal with
    all of the complexity of how Julia syntax is parsed manually.
```